sdk/lib/drivers/debugkd0/debugkd0.h

```diff
@@ -0,0 +1,77 @@
+/*
+ * PROJECT:         ReactOS Kernel
+ * COPYRIGHT:       GPL-2.0+ (https://spdx.org/licenses/GPL-2.0+)
+ * FILE:            lib/drivers/debugkd0/debugkd0.h
+ * PURPOSE:         Kernel phase 0 debug library header
+ * PROGRAMMERS:     Copyright 2020 Vadim Galyant <vgal@rambler.ru>
+ */
+
+#ifndef _DEBUGKD0_H
+#define _DEBUGKD0_H
+
+/* Global control */
+#if DBG
+  #define DBG_KD0 0 // Default disabled
+#else
+  #define DBG_KD0 0 // Default disabled
+#endif
+
+#define COM1 1
+#define COM2 2
+#define COM3 3
+#define COM4 4
+
+#define COM1_ADDR (PUCHAR)0x3F8
+#define COM2_ADDR (PUCHAR)0x2F8
+#define COM3_ADDR (PUCHAR)0x3E8
+#define COM4_ADDR (PUCHAR)0x2E8
+
+#define DEFAULT_COM_BAUD_RATE 19200
+#define DBG_COM_BAUD_RATE 115200
+
+BOOLEAN
+NTAPI
+DbgKdPortInitialize(_In_ ULONG PortNumber OPTIONAL,
+                    _In_ PUCHAR PortAddress OPTIONAL,
+                    _In_ ULONG BaudRate OPTIONAL,
+                    _Out_ PULONG OutPortId);
+
+#if DBG
+
+  #if DBG_KD0
```

Comment:
Phase 0 debugging is rarely needed, so DBG_KD0 defaults to 0.
DBG and DBG_KD0 are used for different purposes.
You can have DEBUGKD0 disabled while DBG is enabled and you can have DEBUGKD0 enabled while DBG is disabled.

Reply:
> you can have DEBUGKD0 enabled while DBG is disabled

That confirms `DBG` must not be checked here, doesn't it?

Reply:
These are the details that each developer (who needs debugging in phase 0) can modify to fit their needs.

Reply:
I don't understand your answer, sorry:
if that very code is what you do want here, then I do not understand the block at line 13 and I do not understand why .c does not have the same `#if`s as here...

Reply:
If you use the debugkd0 library and block at line 13 bothers you, you can use, for example, only one line instead:
#define DBG_KD0 1

This will change depending on the needs of the developer. These are details that are not important in this case. This code will be used very rarely. And only for debugging.

Reply:
> only one line instead

Yes, that would be one way.

>  not important

I don't understand why you like these inconsistencies and I don't understand how how often this is used matters.
Anyway, proceed as you like...